tests/services/nextflowtower/test_ops.py

```diff
@@ -15,21 +18,121 @@ def test_for_an_error_when_the_workspace_attribute_is_missing(patch_os_environ):
-    response = get_response("get_user_workspaces_and_orgs")
-    mock = mocker.patch.object(NextflowTowerOps, "client")
-    response = response["orgsAndWorkspaces"]
-    mock.list_user_workspaces.return_value = response
-    assert ops.workspace_id == 98765
+def test_that_the_workspace_id_attribute_is_accessible(
+    client, config, mocker, get_response
+):
+    response = get_response("get_user_workspaces_and_orgs")["orgsAndWorkspaces"]
+    workspaces_json = [item for item in response if item["workspaceId"]]
+    workspaces = [models.Workspace.from_response(item) for item in workspaces_json]
+
+    mock = mocker.patch.object(NextflowTowerOps, "client", return_value=client)
+    mock.list_user_workspaces.return_value = workspaces
+
+    config.workspace = "foo-bar/project-1"
+    ops = NextflowTowerOps(config)
+    assert ops.workspace_id == 54321
+
 
+def test_for_error_when_the_workspace_id_does_not_exist(
+    client, config, mocker, get_response
+):
+    response = get_response("get_user_workspaces_and_orgs")["orgsAndWorkspaces"]
+    workspaces_json = [item for item in response if item["workspaceId"]]
+    workspaces = [models.Workspace.from_response(item) for item in workspaces_json]
 
-def test_for_error_when_the_workspace_id_does_not_exist(ops, mocker, get_response):
-    # Get rid of relevant entry
-    response = get_response("get_user_workspaces_and_orgs")
-    items = response["orgsAndWorkspaces"]
-    items_filtered = [item for item in items if item["workspaceName"] != "project-2"]
+    mock = mocker.patch.object(NextflowTowerOps, "client", return_value=client)
+    mock.list_user_workspaces.return_value = workspaces
 
-    mock = mocker.patch.object(NextflowTowerOps, "client")
-    mock.list_user_workspaces.return_value = items_filtered
+    config.workspace = "foo-bar/project-3"
+    ops = NextflowTowerOps(config)
     with pytest.raises(ValueError):
         ops.workspace_id
+
+
+def test_that_get_latest_compute_env_handles_multiple_envs(mocked_ops, get_response):
+    # Patch list_compute_envs() to return a list of ComputeEnvSummary objects
+    # This should result in two matching compute environments
+    response = get_response("list_compute_envs")
+    summaries = map(models.ComputeEnvSummary.from_response, response["computeEnvs"])
+    mocked_ops.client.list_compute_envs.return_value = list(summaries)
+
+    # Create function for generating compute environments
+    def generate_compute_env(id, date):
+        compute_env_response = get_response("get_compute_env")["computeEnv"]
+        compute_env = models.ComputeEnv.from_response(compute_env_response)
+        return replace(compute_env, id=id, date_created=datetime.fromisoformat(date))
+
+    # Generate full compute environments
+    mocked_ops.client.get_compute_env.side_effect = [
+        generate_compute_env("5ykJF", "2023-04-26T00:49:49Z"),
+        generate_compute_env("9W2l7", "2023-01-11T13:32:54Z"),
```

Comment:
Same thing here with the isoformat strings
```suggestion
        generate_compute_env("5ykJF", "2023-04-26T00:49:49"),
        generate_compute_env("9W2l7", "2023-01-11T13:32:54"),
```

Reply:
I updated the datetime parsing so that I can consistently use the same format adopted by Tower, which follows RFC 3339 (not ISO 8601).